packages/edit-site/src/components/list/header.js

```diff
@@ -9,10 +9,17 @@ import { __experimentalHeading as Heading } from '@wordpress/components';
  * Internal dependencies
  */
 import AddNewTemplate from '../add-new-template';
+import { store as editSiteStore } from '../../store';
 
 export default function Header( { templateType } ) {
-	const postType = useSelect(
-		( select ) => select( coreStore ).getPostType( templateType ),
+	const { canCreate, postType } = useSelect(
+		( select ) => {
+			const settings = select( editSiteStore ).getSettings();
+			return {
+				postType: select( coreStore ).getPostType( templateType ),
+				canCreate: ! settings?.supportsTemplatePartsMode,
```

Comment:
I would change this to be dependent on whether the theme supports the Template Editing mode. 

So it would not be allowed if the Template Part Editing feature is enabled on its own. But once the Template Editor also gets enabled you gain the ability to create new template parts.

The reason behind this suggestion is that editors already are able to create new template parts from the template editor as it is in Core today. And if they have the Template Editor enabled they are able to use template parts that they manually created. This is not the case if the template editor is not enabled.

Reply:
@mtias, what do you think about this suggestion?